### PR TITLE
Prolific integration

### PR DESCRIPTION
**Pull request by krachwal**
Reading in Prolific redirect querystring parameters and storing them in the session for writing to the file with the results at the end of the trial.

Tested that the values get written with Docker and GitHub Action deployments.

TODO in another PR, redirecting back to Prolific.

### CODE REVIEW

File: template/experiment-server/webserver.py [hunk 1]
```diff
@@ -5,14 +5,15 @@
 from fastapi import FastAPI, Request
 from fastapi.staticfiles import StaticFiles
 from fastapi.responses import JSONResponse, FileResponse, RedirectResponse
+from starlette.middleware.sessions import SessionMiddleware
 
 app = FastAPI()
+app.add_middleware(SessionMiddleware, secret_key="W<(nwGTe8I9MX$UHftcK")
```
Comment (davidbrownell) [inline template/experiment-server/webserver.py]: This secret probably shouldn't be checked in - I'm surprised that it wasn't caught by GitHub.

File: template/experiment-server/webserver.py [hunk 2]
```diff
@@ -22,10 +23,14 @@ def default() -> RedirectResponse:
 
 
 @app.get("/experiment")
-def serve_frontend() -> FileResponse:
+def serve_frontend(request: Request, PROLIFIC_PID: str | None = "", STUDY_ID: str | None = "", SESSION_ID: str | None = "") -> FileResponse:
     """
     Serves static UI
     """
+    request.session["PROLIFIC_PID"] = PROLIFIC_PID
```
Comment (davidbrownell) [inline template/experiment-server/webserver.py:32]: Should these variables be lowercase?

**Review by davidbrownell — CHANGES_REQUESTED** [review]

File: template/experiment-server/webserver.py (hunk 2 above)
Comment (krachwal) [reply template/experiment-server/webserver.py:32]: that is how the querystring parameters are defined... theoretically, sure. urls not being case sensitive, but I was just matching what Prolific calls out.

Comment (krachwal) [reply template/experiment-server/webserver.py:32]: just did a test and it does not work. to be clear, if the querystring has them capitalized or lower case it works as long as the function parameters are capitalized, but if the querystring parameters are capitals and the parameter names in the function header are lower case it does not work and Prolific will send them capitalized.

**Review by davidbrownell — APPROVED** [review]

**Pull request merged**